### ISSUE TITLE
props.intl.formatMessage values not found

### ISSUE THREAD

**Issue by heldrida** [yahoo/react-intl#1113]
### Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

### Expected behavior

'foobar.title': 'xxxx xxxx xxxx {yyyy}',
this.props.intl.formatMessage({ id: 'foobar.title', values: {yyyy: 'barfoo'}})
xxxx xxxx xxxx barfoo

### Current behavior

[React Intl] Error formatting message: "foobar.title" for locale: "en-gb"
Error: The intl string context variable 'yyyy' was not provided to the string 'xxxx xxxx xxxx {yyyy}'

### Your Environment
chrome, macos

| Executable | Version |
├── react-intl@2.4.0
└─┬ react-intl-redux@0.1.0
  └── react-intl@2.4.0  deduped

**Comment by papasmile** [comment]
Hi, @heldrida , `formatMessage` has two parameters, a message descriptor and an object for values: https://github.com/yahoo/react-intl/wiki/API#formatmessage

So you would have:
```
this.props.intl.formatMessage({ id: 'foobar.title' }, { yyyy: 'barfoo' })
```

**Comment by heldrida** [comment]
@papasmile thanks for looking! I missed that, my apologises

**Comment by gabrielmoncea** [comment]
@papasmile I get the same error in Gatsby even though I tried both methods. Am I missing something?

**Comment by smeijer** [comment]
For future Googlers, It looked like I had the same problem. But after a bit research, it turned out to be only the plural strings that weren't parsed.

I think it might be useful if `react-intl` would log a warning in such a case (`can't format string, missing support for `PluralRules`, or something like that).

Any way, I am using `full-icu`, but somehow some api's were still only supported in English.

As soon as I polyfilled Intl on the server as well, problems disappeared:

```js
if (Intl && !Intl.PluralRules) {
  require('@formatjs/intl-pluralrules/polyfill');
  require('@formatjs/intl-pluralrules/dist/locale-data/en');
  require('@formatjs/intl-pluralrules/dist/locale-data/nl');
}

if (Intl && !Intl.RelativeTimeFormat) {
  require('@formatjs/intl-relativetimeformat/polyfill');
  require('@formatjs/intl-relativetimeformat/dist/locale-data/en');
  require('@formatjs/intl-relativetimeformat/dist/locale-data/nl');
}

if (Intl && !Intl.NumberFormat) {
  require('@formatjs/intl-unified-numberformat/polyfill');
  require('@formatjs/intl-unified-numberformat/dist/locale-data/en');
  require('@formatjs/intl-unified-numberformat/dist/locale-data/nl');
}
```

**Comment by nickiaconis** [comment]
For other future Googlers, I was led here by a similar looking problem. Like in @smeijer's case, it only affected plural strings, but in my case `Intl.PluralRules` was available in the execution environment.

After a bit of debugging, I found it was related to the [pound AST element](https://github.com/formatjs/formatjs/blob/81e143e45d72ca96d6a77bba73d5170c1d482c56/packages/intl-messageformat-parser/src/types.ts#L100-L103). For some reason, the formatting code was checking for a value property on pound AST elements, and they don't have a value property.

I resolved this issue by upgrading to the latest version of `react-intl`, which is `3.11.0` at the time of writing. It had previously been pinned to `3.4.0` in my project's `yarn.lock` file.
